components/modal/index.jsx

```diff
@@ -29,6 +29,10 @@ const propTypes = {
 	 * Vertical alignment of Modal.
 	 */
 	align: PropTypes.oneOf(['top', 'center']),
+	/**
+	 * Assistive text for the modal. If not provided, `title` is used.
+	 */
+	assistiveText: PropTypes.string,
```

Comment:
Let's make this an object with a key of `dialog-label`, so that it can be future proof. We may have some more keys to add when we upgrade.

Reply:
I updated it with `dialogLabel`in camel case because I saw that's how other instances of `assistiveText` were defined.  

Also, I noticed that we could put `closeButtonAssistiveText` in there as well, but I'm not sure what the process is for relocating a prop?

Reply:
That's a great idea. There's a `checkProps` function that can run at component mount that will provide a console error in dev mode.

Take a look at [tooltip](https://github.com/salesforce/design-system-react/blob/master/components/popover/check-props.js#L27). This example is the move from `isInline` to `position=relative`.

Then for the actual code, you remove the old prop from the `propTypes` which removes it from the doc site, but you still access it with a `text = newProp || oldProp` privately inside the `render` function, so that it doesn't actually broke anyone's code. Then, we'll officially break it in the future after folks have been given time to change to the new API. @davidlygagnon is working on the something pretty similar right now, too.